surface_potential_analysis/util/plot.py

```diff
@@ -600,6 +612,138 @@ def animate_data_through_list_1d_x(
     return fig, ax, ani
 
 
+def animate_data_through_list_2d_k(
+    basis: TupleBasisLike[*tuple[Any, ...]],
+    data: np.ndarray[tuple[int, int], np.dtype[np.complex128]],
+    axes: tuple[int, int] = (0, 1),
+    idx: SingleStackedIndexLike | None = None,
+    *,
+    ax: Axes | None = None,
+    scale: Scale = "linear",
+    measure: Measure = "abs",
+) -> tuple[Figure, Axes, ArtistAnimation]:
+    """
+    Given data, animate along the given direction.
+
+    Parameters
+    ----------
+    basis : TupleBasisLike
+    data : np.ndarray[tuple[_L0Inv], np.dtype[np.complex_]]
+    axes : tuple[int, int, int], optional
+        plot axes (z, y, z), by default (0, 1, 2)
+    idx : SingleStackedIndexLike | None, optional
+        idx in remaining dimensions, by default None
+    ax : Axes | None, optional
+        plot ax, by default None
+    scale : Scale, optional
+        scale, by default "linear"
+    measure : Measure, optional
+        measure, by default "abs"
+
+    Returns
+    -------
+    tuple[Figure, Axes, ArtistAnimation]
+    """
+    fig, ax = get_figure(ax)
+
+    frames: list[list[QuadMesh]] = []
+    i = 0
+    for data_i in data:
+        if i == 0:
```

Comment:
this is no longer necessary if you use _has_colorbar(), but for future reference using enumerate(data) is preferred over i=0, i+=1